content/en/guide/v10/typescript.md

```diff
@@ -194,16 +189,14 @@ export class Button extends Component {
 }
 ```
 
-You can restrict event handlers by adding a type annotation for `this` to the function signature as the first argument. This argument will be erased after transpilation.
+You can provide a more specific type than `EventTarget` by using [this parameters](https://www.typescriptlang.org/docs/handbook/functions.html#this-parameters):
 
 ```tsx
 export class Button extends Component {
-  // Adding the this argument restricts binding
+  // Define `this parameters`
   handleClick(this: HTMLButtonElement, event: MouseEvent) {
     event.preventDefault();
-    if (event.target instanceof HTMLElement) {
-      console.log(event.target.localName); // "button"
-    }
+    console.log(this.tagName); // "BUTTON"
   }
```

Comment:
I think this example should use `this`.
Since `this` is `HTMLButtonElement`, `instanceof` is unnecessary.
Another example uses `tagName`.

Reply:
I agree, this is much more concise and showcases this binding much better 👍